File: LogisimPlus/LogisimPlus/logisim/gui/appear/ClipboardContents.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// To produce customized conversions, purchase a Premium Edition license:
// https://www.tangiblesoftwaresolutions.com/product-details/java-to-csharp-converter.html
// ====================================================================================================

using System.Collections.Generic;
using System.Linq;

/* Copyright (c) 2011, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

namespace logisim.gui.appear
{

	using CanvasObject = draw.model.CanvasObject;
	using Direction = logisim.data.Direction;
	using Location = logisim.data.Location;

	internal class ClipboardContents
	{
		internal static readonly ClipboardContents EMPTY = new ClipboardContents(Enumerable.Empty<CanvasObject>(), null, null);

		private ICollection<CanvasObject> onClipboard;
		private Location anchorLocation;
		private Direction anchorFacing;

		public ClipboardContents(ICollection<CanvasObject> onClipboard, Location anchorLocation, Direction anchorFacing)
		{
			this.onClipboard = (new List<CanvasObject>(onClipboard)).AsReadOnly();
			this.anchorLocation = anchorLocation;
			this.anchorFacing = anchorFacing;
		}

		public virtual ICollection<CanvasObject> Elements
		{
			get
			{
				return onClipboard;
			}
		}

		public virtual Location AnchorLocation
		{
			get
			{
				return anchorLocation;
			}
		}

		public virtual Direction AnchorFacing
		{
			get
			{
				return anchorFacing;
			}
		}
	}

}

File: logisim/app/src/main/java/logisim/gui/appear/ClipboardContents.java
/* Copyright (c) 2011, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

package logisim.gui.appear;

import java.util.Collection;
import java.util.Collections;
import java.util.List;

import draw.model.CanvasObject;
import logisim.data.Direction;
import logisim.data.Location;

class ClipboardContents {
	static final ClipboardContents EMPTY = new ClipboardContents(Collections.emptySet(), null, null);

	private Collection<CanvasObject> onClipboard;
	private Location anchorLocation;
	private Direction anchorFacing;

	public ClipboardContents(Collection<CanvasObject> onClipboard, Location anchorLocation, Direction anchorFacing) {
		this.onClipboard = List.copyOf(onClipboard);
		this.anchorLocation = anchorLocation;
		this.anchorFacing = anchorFacing;
	}

	public Collection<CanvasObject> getElements() {
		return onClipboard;
	}

	public Location getAnchorLocation() {
		return anchorLocation;
	}

	public Direction getAnchorFacing() {
		return anchorFacing;
	}
}
